grpc.go

```diff
@@ -993,13 +993,31 @@ func (a *agentGRPC) RemoveContainer(ctx context.Context, req *pb.RemoveContainer
 		if err := ctr.removeContainer(); err != nil {
 			return emptyResp, err
 		}
+
+		// Find the sandbox storage used by this container
+		for _, path := range ctr.mounts {
+			if _, ok := a.sandbox.storages[path]; ok {
+				if err := a.sandbox.unsetAndRemoveSandboxStorage(path); err != nil {
+					return emptyResp, err
+				}
+			}
+		}
 	} else {
 		done := make(chan error)
 		go func() {
 			if err := ctr.removeContainer(); err != nil {
 				done <- err
 			}
 
+			// Find the sandbox storage used by this container
+			for _, path := range ctr.mounts {
+				if _, ok := a.sandbox.storages[path]; ok {
+					if err := a.sandbox.unsetAndRemoveSandboxStorage(path); err != nil {
+						done <- err
```

Comment:
Here you need to return and close(done), the `done` is only a channel without buffer which can block if there's no receiver. so the go routine could leak here.

Reply:
channel is getting closed here, https://github.com/kata-containers/agent/pull/293/files#diff-727ccd4217b7f335e689be03bfb69e9bR1021

Reply:
Makes sense. Thanks.